src/DynamoCore/Models/DynamoModel.cs

```diff
@@ -1771,6 +1789,30 @@ private void RegisterHomeWorkspace(HomeWorkspaceModel newWorkspace)
             };
         }
 
+        private HashSet<string> OnCollectingAssembliesUsed()
+        {
+            var assemblies = new HashSet<string>();
+            foreach(var node in CurrentWorkspace.Nodes)
+            {
+                IEnumerable<FunctionDescriptor> descriptors = LibraryServices.GetAllFunctionDescriptors(node.CreationName.Split('@')[0]);
```

Comment:
document this

Reply:
tiny nit pick, because this is a bit wide, it doesn't view well on github.

Reply:
hmm, won't this only work for ZeroTouch nodes, what about nodeModel nodes inside of packages?

Reply:
Right, good catch. Works for ZeroTouch and NodeModel now.